src/components/ui/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Text, TouchableOpacity, View } from "react-native";
+import DateTimePickerModal from "react-native-modal-datetime-picker";
+import { Icon } from "@app/components/ui";
+import { useStyles } from "./styles";
+import dayjs from "dayjs";
+
+interface DateTimeInputProps {
+  id: string;
+  value?: string; // GMT format
+  onConfirm: (date: string) => void;
+  errorMessage?: string;
+  placeholder?: string;
+}
+
+const DateTimeInput: React.FC<DateTimeInputProps> = ({
+  id,
+  value,
+  onConfirm,
+  errorMessage,
+  placeholder,
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [currentDate, setCurrentDate] = useState<Date | undefined>(undefined);
+  const styles = useStyles({ date: currentDate });
+
+  useEffect(() => {
+    if (value) {
+      setCurrentDate(new Date(value));
+    }
+  }, [value]);
+
+  const showDatePicker = useCallback(() => {
+    setIsVisible(true);
+  }, []);
+
+  const hideDatePicker = useCallback(() => {
+    setIsVisible(false);
+  }, []);
+
+  const handleConfirm = useCallback(
+    (date: Date) => {
+      onConfirm(date.toUTCString());
+      setCurrentDate(date);
+      hideDatePicker();
+    },
+    [hideDatePicker, onConfirm],
+  );
+
+  const displayText = useMemo(() => {
+    if (currentDate) {
+      return dayjs(currentDate).format("DD MMMM [at] hh:mm a");
+    } else {
+      return placeholder || "Enter date";
+    }
+  }, [placeholder, currentDate]);
+
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity onPress={showDatePicker}>
+        <View style={styles.inputContainer} data-testid={`${id}-date-input`}>
+          <View>
+            <Icon id="date-picker" name="calendar" type="ionicon" />
+          </View>
+          <View style={styles.input}>
+            <Text style={styles.inputText}>{displayText}</Text>
+          </View>
+        </View>
+      </TouchableOpacity>
+      <View style={styles.errorContainer}>
+        {errorMessage && (
```

Comment:
should errorMessage be wrapped above the View component?

Reply:
I was matching the styles of the react-native-elements input component. I think they still have the view there even without the error text

Reply:
hm but the View doesnt need to show unless there's an error message right?